packages/react-static/src/static/webpack/webpack.config.dev.js

```diff
@@ -35,7 +35,7 @@ export default function({ config }) {
       filename: '[name].js', // never hash dev code
       chunkFilename: 'templates/[name].js',
       path: DIST,
-      publicPath: process.env.REACT_STATIC_ASSETS_PATH || '/',
+      publicPath: process.env.REACT_STATIC_PUBLIC_PATH || '/',
```

Comment:
👋 wouldn't this be missing a leading slash ? Otherwise this ends up creating relative URLs - which I think is what would be causing the double "dist/dist" mentioned in https://github.com/nozzle/react-static/issues/1141#issuecomment-499085134.

Reply:
(sorry, meant to comment on the `webpack.config.prod.js` file)

Reply:
Done some more testing around this and seems like `process.env.REACT_STATIC_PUBLIC_PATH` would benefit from being absolute everywhere, maybe it would be worth to just update it here: https://github.com/nozzle/react-static/blob/master/packages/react-static/src/static/getConfig.js#L133 (much like `assetPath` is made absolute below) ?

Reply:
Otherwise if we set `siteRoot`, we'll get an absolute URL - we only get a relative URL if using `basePath` without `siteRoot`.

Reply:
@nddery I think you're on to something there. Were you able to investigate this any more?

Reply:
Along with the change proposed in this pull request, what fixed things for us was to update the logic around setting the `publicPath` in `getConfig.js` to:

```diff
--- a/packages/react-static/src/static/getConfig.js
+++ b/packages/react-static/src/static/getConfig.js
@@ -127,10 +127,12 @@ export function buildConfig(state, config = {}) {
     siteRoot = cutPathToRoot(config.stagingSiteRoot || '/', '$1')
     basePath = cleanSlashes(config.stagingBasePath)
   } else {
-    siteRoot = cutPathToRoot(config.siteRoot, '$1')
+    siteRoot = cutPathToRoot(config.siteRoot || '/', '$1')
     basePath = cleanSlashes(config.basePath)
   }
-  const publicPath = `${cleanSlashes(`${siteRoot}/${basePath}`)}/`
+  const publicPath = `${cleanSlashes(`${siteRoot}/${basePath}`, {
+    leading: false,
+  })}/`
   let assetsPath = cleanSlashes(config.assetsPath || paths.assets)
   if (assetsPath && !isAbsoluteUrl(assetsPath)) {
     assetsPath = `/${cleanSlashes(`${basePath}/${assetsPath}`)}/`
```

This way, `publicPath`/`process.env.REACT_STATIC_PUBLIC_PATH` is always absolute.

This revealed another issue with route info prefetching for the homepage, which seems to come from  the [`getRoutePath`](https://github.com/nozzle/react-static/blob/master/packages/react-static/src/browser/utils/index.js#L64) browser utility function. Seems like the regex to remove the `basePath` doesn't seem to do work when `routePath === process.env.REACT_STATIC_BASE_PATH` (ie.: when `routePath` is pointing to the homepage) - ending up trying to fetch `/basePath/basePath/routeInfo.json`. I got to fix the 404 by adding a `if` statement for that case but then I can't navigate to the homepage (stuck on "loading"). I can open another issue for this if it make sense.

```diff
--- a/packages/react-static/src/browser/utils/index.js
+++ b/packages/react-static/src/browser/utils/index.js
@@ -63,7 +63,11 @@ export function pathJoin(...paths) {
 // resemble the same string as passed in the static.config.js routes
 export function getRoutePath(routePath) {
   // Detect falsey paths and the root path
-  if (!routePath || routePath === '/') {
+  if (
+    !routePath ||
+    routePath === '/' ||
+    routePath === process.env.REACT_STATIC_BASE_PATH
+  ) {
     return '/'
   }
```

Happy to help - just let me know how to, I'm still trying to wrap my head around all the moving parts!

Reply:
@tannerlinsley does the above make sense to you ? Would you like to see a pull request with these changes or can you think of any issues this change would create ?

Reply:
@nddery If you create the PR we can run through it together!

Reply:
I've implemented those changes in https://github.com/react-static/react-static/pull/1272 - although tests are failing right now..